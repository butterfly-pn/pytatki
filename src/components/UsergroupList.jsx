import React, { Component } from "react";
import style from "../scss/UsergroupList.scss";

export default class UsergroupList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      usergroups: []
    };
    this.getUsergroups();
  }

  getUsergroups = () => {
    const siteUrl = this.props.siteUrl;
    const that = this;
    return fetch(siteUrl + "/api/?query={getToken}")
      .then(response => response.json())
      .then(res => {
        if (/\d/.test(res.data.getToken)) {
          return res.data.getToken;
        }
        alert(res.data.getToken);
      })
      .then(token =>
        fetch(
          siteUrl + '/api?query={getUsergroups(access_token:"' + token + '")}'
        )
      )
      .then(response => response.json())
      .then(myJson => JSON.parse(myJson.data.getUsergroups))
      .then(function(innerJson) {
        let usergroups = [];
        for (const usergroup of innerJson) {
          let object = {};
          object["key"] = usergroup.idusergroup;
          object["name"] = usergroup.name;
          object["color"] = usergroup.color;
          object["imagePath"] = usergroup.image_path;
          usergroups.push(object);
        }
        return usergroups;
      })
      .then(plainGroups => {
        let groups = [];
        for (const group of plainGroups) {
          groups.push(
            <li key={group.key}>
              <button
                className="btn"
                onClick={this.props.updateUsergroup}
                id={group.key}
              >
                {group.name}
              </button>
            </li>
          );
        }
        that.setState({
          usergroups: groups
        });
      })
      .catch(error => console.log(error));
  };

  render() {
    return (
      <div className={style.sidebar}>
        <p>Klasy</p>
        <ul>{this.state.usergroups}</ul>
      </div>
    );
  }
}
