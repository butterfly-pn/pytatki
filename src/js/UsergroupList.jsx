import React, { Component } from 'react'

export default class UsergroupList extends Component {
  constructor(props) {
    super(props)
    this.state = {
      usergroups: []
    }
    this.getUsergroups();
  }
  
  getUsergroups = () => {
    const siteUrl = this.props.siteUrl;
    const that = this;
    return fetch(siteUrl + '/api?query={getToken}')
      .then(response => response.json())
      .then(res => res.data.getToken)
      .then(token => fetch(siteUrl + '/api?query={getUsergroups(access_token:"' + token + '")}'))
      .then(response => response.json())
      .then(myJson => JSON.parse(myJson.data.getUsergroups))
      .then(function (innerJson) {
        let usergroups = [];
        for (const usergroup of innerJson) {
          let object = {};
          object["key"] = usergroup.idusergroup;
          object["name"] = usergroup.name;
          object["color"] = usergroup.color;
          object["imagePath"] = usergroup.image_path;
          usergroups.push(object);
        };
        return usergroups
      })
      .then(plainGroups => {
        let groups = []
        for (const group of plainGroups) {
          groups.push(<h1 onClick={this.props.updateUsergroup} key={group.key} id={group.key}>{group.name}</h1>);
        }
        that.setState({
          usergroups: groups
        });
      })
      .catch(error => console.log(error));
  }

  render() {
    return (
      <div>
        UsergroupList
      {this.state.usergroups}
      </div>
    )
  }
}