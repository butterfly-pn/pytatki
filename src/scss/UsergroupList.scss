.sidebar{
    position: fixed;
    height: 100vh;
    width: 150px;
    vertical-align: top;
}