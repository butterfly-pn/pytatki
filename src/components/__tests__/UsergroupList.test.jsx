import React from 'react';
import UsergroupList from '../UsergroupList.jsx';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Snapshot test', () => {
    beforeEach(() => {
        window.fetch = jest.fn((link) => {
            if (link.includes('getToken')) {
                return Promise.resolve({
                    status: 200,
                    json: () => Promise.resolve({
                        "data": {
                            "getToken": "randomString2"
                        },
                    })
                })
            } else if (link.includes('/api?query={getUsergroups(access_token:"randomString2")}')) {
                return Promise.resolve({
                    status: 200,
                    json: () => Promise.resolve({
                        "data": {
                            "getUsergroups": '[{"idusergroup":1,"name":"grupa"}, {"idusergroup":2,"name":"grupa42"}]'
                        },
                    })
                })
            }
        })
    })
    const delay = () => new Promise(res => setTimeout(res));

    it('renders as expected', async () => {
        const wrapper = shallow(<UsergroupList />)
        await  delay();
        expect(wrapper).toMatchSnapshot();

    })
})