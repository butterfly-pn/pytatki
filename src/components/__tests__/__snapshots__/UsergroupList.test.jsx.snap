// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders as expected 1`] = `
<div
  className="sidebar"
>
  <p>
    Klasy
  </p>
  <ul>
    <li
      key="1"
    >
      <button
        className="btn"
        id={1}
      >
        grupa
      </button>
    </li>
    <li
      key="2"
    >
      <button
        className="btn"
        id={2}
      >
        grupa42
      </button>
    </li>
  </ul>
</div>
`;
